### PR TITLE
SYCL IJ without unified memory

### PR DESCRIPTION
**Pull request by waynemitchell**
This PR ports a few remaining routines required for running the ij.c driver without unified memory using the sycl backend (for some limited subset of options). Ported routines are:
1. IJ vector set/add values and assembly
2. Routines in par_coarse_parms_device.c
3. hypre_ParCSRComputeL1Norms()

### CODE REVIEW

**Comment by waynemitchell** [comment]
Regression tests pass on tux, lassen, and jlse.

File: src/test/TEST_ij/solvers.saved.arcticus [hunk 1]
```diff
@@ -134,8 +134,8 @@ Iterations = 19
 Final Relative Residual Norm = 3.502389e-09
 
 # Output file: solvers.out.107
-Iterations = 34
-Final Relative Residual Norm = 8.725157e-09
+Iterations = 29
```
Comment (ulrikeyang) [inline src/test/TEST_ij/solvers.saved.arcticus:137]: Convergence improved a lot here. Was there a bug before?

**Review by ulrikeyang — APPROVED** [review]
This looks good as far as I can tell with limited SYCL knowledge .

**Review by rfalgout — APPROVED** [review]
Looks good to me too!

File: src/test/TEST_ij/solvers.saved.arcticus (hunk 1 above)
Comment (waynemitchell) [reply src/test/TEST_ij/solvers.saved.arcticus:137]: Yes, I think there was a previous bug, though I don't think it was fixed in this PR. When I reran master on this test I got the same (29 iteration) result shown here. So I guess I just hadn't updated the .saved file previously.

**Review by liruipeng — APPROVED** [review]
Looks good to me. Again, I trusted @waynemitchell on the sycl implementation details. Thanks!

**Pull request merged**